v-next/hardhat/src/internal/cli/telemetry/sentry/anonymizer.ts

```diff
@@ -0,0 +1,401 @@
+import type { Event, Exception, StackFrame, Stacktrace } from "@sentry/node";
+
+import * as path from "node:path";
+
+import {
+  findClosestPackageJson,
+  PackageJsonNotFoundError,
+} from "@ignored/hardhat-vnext-utils/package";
+// This file is executed in a subprocess, and it is always run in this context.
+// Therefore, it is acceptable to avoid using dynamic imports and instead import all necessary modules at the beginning.
+import * as czech from "ethereum-cryptography/bip39/wordlists/czech.js";
+import * as english from "ethereum-cryptography/bip39/wordlists/english.js";
+import * as french from "ethereum-cryptography/bip39/wordlists/french.js";
+import * as italian from "ethereum-cryptography/bip39/wordlists/italian.js";
+import * as japanese from "ethereum-cryptography/bip39/wordlists/japanese.js";
+import * as korean from "ethereum-cryptography/bip39/wordlists/korean.js";
+import * as simplifiedChinese from "ethereum-cryptography/bip39/wordlists/simplified-chinese.js";
+import * as SPANISH from "ethereum-cryptography/bip39/wordlists/spanish.js";
+import * as traditionalChinese from "ethereum-cryptography/bip39/wordlists/traditional-chinese.js";
+import { either } from "fp-ts";
+
+interface WordMatch {
+  index: number;
+  word: string;
+}
+
+const ANONYMIZED_FILE = "<user-file>";
+const ANONYMIZED_MNEMONIC = "<mnemonic>";
+const MNEMONIC_PHRASE_LENGTH_THRESHOLD = 7;
+const MINIMUM_AMOUNT_OF_WORDS_TO_ANONYMIZE = 4;
+
+export class Anonymizer {
+  readonly #configPath?: string;
+
+  constructor(configPath?: string) {
+    this.#configPath = configPath;
+  }
+
+  /**
+   * Given a sentry serialized exception
+   * (https://develop.sentry.dev/sdk/event-payloads/exception/), return an
+   * anonymized version of the event.
+   */
+  public async anonymize(event: any): Promise<either.Either<string, Event>> {
```

Comment:
Let's replace this with something like:

```ts
type AnonymizeResult = {success: true, Event} | {success: false, error:string}
```

so that we don't add `fp-ts` as a dependency.